src/api/consumer/lib/switch.ts

```diff
@@ -26,15 +26,21 @@ export default async function switchAction(
 
 async function populateSwitchProps(consumer: Consumer, switchProps: SwitchProps) {
   const lanes = await consumer.scope.listLanes();
+  const isDefaultLane = switchProps.laneName === DEFAULT_LANE;
+
   const localLane = lanes.find((lane) => lane.name === switchProps.laneName);
 
-  if (localLane) {
+  if (isDefaultLane || localLane) {
     populatePropsAccordingToLocalLane();
   } else {
     await populatePropsAccordingToRemoteLane();
   }
 
   async function populatePropsAccordingToRemoteLane() {
+    if (switchProps.laneName.includes(DEFAULT_LANE)) {
```

Comment:
if I have a remote lane `my-main`, I believe this condition will return true, although it's a valid lane-name to switch to.
Maybe this code should be a few lines later after parsing the lane-id to scope and name, and then it's easy to check whether it's equal.